### ISSUE TITLE
2.1 Final Stuff

### ISSUE THREAD

**Issue by Jebbs** [Jebbs/DSFML#158]
The milestone for 2.1 is slightly misleading. 

I have a branch on my own computer that is basically finished save for a couple of very small things. Once the code is finished, I will be knocking out tutorials and getting the website ready.

That said, I'm sure that I will need to do other things related to the release of 2.1 and here's all the things that I can think of:
* Create packages for each OS with shared libs, static libs, and source
* Write a change log 
* tag the repo and create a release
* post about it?

Please feel free to add things in here that you'd like me to do related to the release!

**Comment by aubade** [comment]
Any updates on this? :)

**Comment by Jebbs** [comment]
Not really, unfortunately. I have finals next week, so you can expect it sometime this summer. I'm going to shoot for July.

2.2 will take considerably less time too since I'll also be putting systems in place to make all this stuff easier to do in the future.

**Comment by Jebbs** [comment]
Just got all unit tests passing on all systems after upgrading the SFML code in the back end to 2.1. I'll take a couple of days to do a little bit of cleaning up, and then upload everything and merge into master. By that point, all of the code should be 100% finished unless anyone else can find anything that needs to be fixed before release.

Then I'll do the tutorials, documentation, and packages. By that point, we should be ready for the official release of 2.1!

**Comment by luke5542** [comment]
Hey, just wonder what, at this point, is limiting the progress on the 2.1 update? If tutorials are still part of that, I'll gladly get started on some more translations of SFML's tutorials (which I should probably do already).

**Comment by Jebbs** [comment]
In terms of actual work to be done, yes mostly just tutorials. I think I wanted to do a little work on getting online documentation a little nicer too, but that's pretty much it.

Unfortunately I had an unexpected work trip so I was out of the county for the last three weeks. I literally just returned today.

**Comment by luke5542** [comment]
Ah, that makes sense.

Is there a particular tutorial setup you were looking for? I've just got a git repo of markdown files setup, but it might be more efficient for future tutorials (in terms of linking between the pages, and any other differences that might occur) if a more long-term final setup was settled upon sooner.

**Comment by luke5542** [comment]
Okay, so I'm basically finished translating the tutorials from SFML to DSFML, over in [this repo](https://github.com/luke5542/DSFML-Tutorials), and would like some feedback on them and if I missed anything, etc. I think the next step is, as I mentioned previously, finding a more permanent home for them.

My thoughts are to either add them to this repo, or I can try and compile them all to HTML (possibly grabbing in any CSS that currently exists for dsfml.com) and push them to the gh pages branch.

**Comment by Jebbs** [comment]
Holy crap. This is huge. I can't thank you enough for this.

I think it would be nice to have them in the website. I should be able to have feedback on the next couple of days. The only ones I would need to do myself are some updated build tutorials.

I'm going to try to finish any code related stuff by this weekend. Once that's fine I'll announce it in the newsgroup to see if we can get any last minute bug reports. Then it's pretty much cleaning up the website, do binary packages, and release.

**Comment by luke5542** [comment]
You're most welcome! I felt it was only necessary to give something in return, as I used this library in my school project to get some pretty excellent marks.

After a quick Google search, it seems to me like it would be pretty easy to just add them into the gh_pages branch and have [GitHub parse the markdown](https://help.github.com/articles/using-jekyll-with-pages/). I'm not yet certain how well the styling would work, but hopefully that's not too much of a problem.

That sounds like a good plan! It will be nice to see this release happen.

**Comment by Jebbs** [comment]
Uhg, finally got some time to review some of these. They look pretty good. I only saw a few minor things that need changing fir the website like links and such. I can handle converting them to html (going to try out Pandoc) and getting them set up on the site.

**Comment by Jebbs** [comment]
So I think all the code is finished. I have some extra time off this week and should have that confirmed on Tuesday.

All the tutorials are now in html form, I just need to get them wrapped in the same way the rest of the pages on the site are done. 

I'm going to aim for having everything finished by the end of the week.

**Comment by Jebbs** [comment]
PAX was last weekend. Whoops. :P 

All tutorials @luke5542 completed are now "finished" and put up on the [website](http://dsfml.com/tutorials.html)! I'm going to keep working on stuff tonight so we'll see what all I get done.

**Comment by Jebbs** [comment]
OK, everyone. Tomorrow should be the day.

Aside from bumping up the version in config.d, I'm calling the front end finished. I'm currently cleaning up the back end, but it shouldn't take much longer.

I'm hoping that the only things I have left tomorrow are the build tutorials, but that won't take too long as I am more or less doing part of it right now (double checking that it builds and runs on all 3 systems).

Sorry it took so long. :P

**Comment by Jebbs** [comment]
I ran into some issues getting stuff built for downloading. What I will do today is merge all the code into master and tag it though. Assuming my build machines stop hating me, I'll be able to finish getting the download packages done by the weekend.

**Comment by Jebbs** [comment]
Just kidding. Something came up so I won't have the build tutorials finished tonight. :(

**Comment by Jebbs** [comment]
Today is the day!

The only things I have left to do is:
+ merge the rc branches back into master
+ update my compilers for final builds
+ build final packages
+ put the links to the final packages on the download page
+ Post announcement on the forums
+ pray nothing goes horribly wrong

Let's do this!

**Comment by Jebbs** [comment]
Well, god damn it.

Windows is giving me issues (something with VS2015), so I have to push this back again. I won't get to work on it again until Tuesday due to studies.

**Comment by Jebbs** [comment]
http://forum.dlang.org/post/lvpwnztnatnmttpybfud@forum.dlang.org
